### PR TITLE
shot-scraper: init at 1.1.1

### PR DESCRIPTION
**Pull request by techknowlogick**
###### Description of changes

https://github.com/simonw/shot-scraper/releases/tag/1.0

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

Related to #81418

### CODE REVIEW

**Comment by fabaff** [comment]
Result of `nixpkgs-review pr 196974` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>shot-scraper (python310Packages.shot-scraper)</li>
    <li>python39Packages.shot-scraper</li>
  </ul>
</details>

File: pkgs/development/python-modules/shot-scraper/default.nix [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+, click
+, click-default-group
+, pytestCheckHook
+, hypothesis
+, pytest-mock
+, playwright
+, pyyaml
+}:
+
+buildPythonPackage rec {
```
Comment (fabaff) [inline pkgs/development/python-modules/shot-scraper/default.nix:14]: According to the documentation:

> A command-line utility for taking automated screenshots of websites

As it's not a library, it should be handled as a tool. 

```suggestion
python3.pkgs.buildPythonApplication rec {
```

Perhaps move it to `tools/graphics/`

File: pkgs/development/python-modules/shot-scraper/default.nix [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+  propagatedBuildInputs = [
+    click
+    click-default-group
+    playwright
+    pyyaml
+  ];
+
+  # skip tests due to network access
+#   doCheck = false;
+
```
Comment (fabaff) [inline pkgs/development/python-modules/shot-scraper/default.nix:35]: Stale?

File: pkgs/development/python-modules/shot-scraper/default.nix (hunk 1 above)
Comment (techknowlogick) [reply pkgs/development/python-modules/shot-scraper/default.nix]: Thanks @fabaff, I've moved this to the other directory :)

File: pkgs/development/python-modules/shot-scraper/default.nix (hunk 2 above)
Comment (techknowlogick) [reply pkgs/development/python-modules/shot-scraper/default.nix]: uncommented, thanks :)

**Comment by fabaff** [comment]
Result of `nixpkgs-review pr 196974` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>shot-scraper</li>
  </ul>
</details>

**Review by fabaff — APPROVED** [review]
- [x] package path fits guidelines
- [x] package name fits guidelines
- [x] package version fits guidelines
- [x] `meta.description` is set and fits guidelines
- [x] `meta.license` fits upstream license
- [x] `meta.maintainers` is set
- [x] package builds (see https://github.com/NixOS/nixpkgs/pull/196974#issuecomment-1475806257)
- [ ] all depending packages build
- [x] tested on x86_64

```bash
...ew/pr-196974]$ ./results/shot-scraper/bin/shot-scraper 
Usage: shot-scraper shot [OPTIONS] URL
Try 'shot-scraper shot --help' for help.

Error: Missing argument 'URL'.
```
Looks good to me.

**Pull request merged**